### PR TITLE
updated CONTRIBUTING.md, just made it more fun

### PR DESCRIPTION
**Pull request by 0Armaan025**
### All Submissions:

- [✔️ ] Have you followed the guidelines stated in the [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [✔️] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls ) for the desired change?

### Type of change

Please delete options that are not relevant.
- [✔️ ] This change requires a documentation update

INFO:

Hey there!
First, if you are reading this, you are amazing!

I am Armaan and actively taking part in Hacktoberfest and HackSquad, I came across this amazing repo to contribute and found out "CONTRIBUTING.md" as a bit plain ( my opinion ), and thought why not contribute to make it more fun and that's what I did!

There have been no code changes, just the docs update as I thought it was not too engaging, nothing else!

If the maintainers/repo owners want to keep it the same as before, I'm sorry for this unnecessary PR and taking up your precious time, you can definitely have a look at it, if you liked it, you can merge it otherwise, this PR can be closed and I apologise if there are any mistakes.

Additional INFO:  Starred✔️⭐

Thanks and regards,
Armaan.

### CODE REVIEW

**Review by masenf — APPROVED** [review]
Cool, thanks for contributing

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -57,42 +55,42 @@ All the changes you make to the repository will be reflected in your running app
 
 ## 🧪 Testing and QA
 
-Any feature or significant change added should be accompanied with unit tests.
+Any feature or significant change added should be accompanied by unit tests.
 
-Within the 'test' directory of Reflex you can add to a test file already there or create a new test python file if it doesn't fit into the existing layout.
+Within the 'test' directory of Reflex, you can add to an existing test file or create a new test Python file if it doesn't fit the current layout.
```
Comment (Lendemor) [inline CONTRIBUTING.md:60]:
```suggestion
Within the 'tests' directory of Reflex, you can add to an existing test file or create a new test Python file if it doesn't fit the current layout.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -57,42 +55,42 @@ All the changes you make to the repository will be reflected in your running app
 
 ## 🧪 Testing and QA
```
Comment (Lendemor) [inline CONTRIBUTING.md:56]: I don't think we need the `## Testing and QA` title twice.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,53 +1,51 @@
...
-- Any edge cases or potential problem areas.
-- Any interactions between different parts of the code.
+#### What to Unit Test?
+- Any new feature or significant change.
+- Edge cases or potential problem areas.
+- Interactions between different parts of the code.
 
-Now Init/Run
-``` bash
+Now, Init and Run:
```
Comment (Lendemor) [inline CONTRIBUTING.md:47]: I think this part was repeated wrongly, should probably be removed as it already appear on line 33.

**Review by Lendemor — CHANGES_REQUESTED** [review]
A few changes are needed, but I love it, thanks.

**Comment by masenf** [comment]
Please resolve conflicts and reopen if you still wish to contribute.
